odl/test/discr/lp_discr_test.py

```diff
@@ -189,6 +174,66 @@ def test_uniform_discr_init_complex(tspace_impl):
     assert discr.dtype == discr.tspace.default_dtype(odl.ComplexNumbers())
 
 
+def test_uniform_discr_empty():
+    """Check if empty spaces behave as expected and all methods work."""
+    discr = odl.uniform_discr([], [], ())
+
+    assert discr.interp == 'nearest'
+    assert discr.axis_labels == ()
+    assert discr.tangent_bundle == odl.ProductSpace(field=odl.RealNumbers())
+    assert discr.complex_space == odl.uniform_discr([], [], (), dtype=complex)
+    assert hash(discr) == hash(discr)
+    assert repr(discr) != ''
+
+    elem = discr.element(1.0)
```

Comment:
waaaait a minute, is this truly what we expect from an empty space? Shouldn't it have size 0?

Reply:
It has, but it should have size 1. That would be the equivalent to NumPy's handling of 0D arrays:
```python
>>> a = np.array(1.0)
>>> a.ndim
0
>>> a.size
1
```
I'll go fix it.

Reply:
Wow, i guess you are right :O keep it this way then (with appropriate bug-fixes)